### PR TITLE
Fix multicast Port_Group update at pod creation/deletion.

### PR DESCRIPTION
**Pull request by dceara**
Whenever a pod is added/removed from a namespace 'ns', if multicast is
enabled in the namespace, also update the associated Port_Group
mcastPortGroup-<ns> otherwise multicast traffic will not be allowed from
the pod.

Whenever a namespace is created with existing pods, add the pods to the
default multicast deny policy.

Reported-at: https://bugzilla.redhat.com/1772579
Fixes: c3def15ad986 ("Add multicast support.")
Signed-off-by: Dumitru Ceara <dceara@redhat.com>

### CODE REVIEW

**Comment by dceara** [comment]
@dcbw @danwinship PTAL.

**Comment by dcbw** [comment]
@dceara could you rebase? Another PR added policy_test.go but it should be pretty close in structure to yours... otherwise LGTM

**Comment by dceara** [comment]
> @dceara could you rebase? Another PR added policy_test.go but it should be pretty close in structure to yours... otherwise LGTM

@dcbw Ack, I just rebased. While doing that I found another issue that was uncovered by the new `LooseCompareFakeExec` tests: Whenever a namespace is created with existing pods, the pods weren't added to the default multicast deny policy. I fixed that case too and it's covered by unit tests now.

**Comment by dceara** [comment]
@dcbw Rebased against latest master and fixed new conflicts in the unit tests.

**Comment by dcbw** [comment]
@dceara gonna need another rebase... sorry about that.

File: go-controller/pkg/ovn/policy.go [hunk 1]
```diff
@@ -358,6 +358,12 @@ func (oc *Controller) getMulticastACLMatch(ns string) string {
 	return "ip4.src == $" + nsAddressSet + " && ip4.mcast"
 }
 
+// Returns the multicast port group name and hash for namespace 'ns'.
+func (oc *Controller) getMulticastPortGroup(ns string) (string, string) {
```
Comment (dcbw) [inline go-controller/pkg/ovn/policy.go]: @dceara don't need `(oc *Controller)` here since it doesn't use any members of `oc`

Comment (dceara) [reply go-controller/pkg/ovn/policy.go]: @dcbw, ok, fixed and also rebased on the latest master.

**Comment by dceara** [comment]
@dcbw Thanks!

**Pull request merged**